workers/main/src/index.ts

```diff
@@ -37,10 +37,12 @@ export async function run(): Promise<void> {
   }
 }
 
-export function handleRunError(err: unknown): never {
-  logWorkerError('main', err);
+export function handleRunError(error: unknown): never {
+  logger.error(
+    `Error in main worker: ${error instanceof Error ? error.message : String(error)}`,
+  );
   setTimeout(() => process.exit(1), 100);
-  throw err;
+  throw error;
```

Comment:
Do you think you need to rethrow here. This is the highest order error handler, maybe it needs to output error and exit?